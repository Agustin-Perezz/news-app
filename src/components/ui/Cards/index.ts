export * from './ListCards';
export * from './customStyleCard';
