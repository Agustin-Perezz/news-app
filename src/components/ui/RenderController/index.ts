export * from './LoadingSkeleton';
export * from './RenderController';
