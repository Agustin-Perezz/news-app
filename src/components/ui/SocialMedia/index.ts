export * from './SocialMedia';
export * from './data-media';
