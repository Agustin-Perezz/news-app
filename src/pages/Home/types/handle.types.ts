export type HandleClickCountryProps = {
  handleClickCountry: (codeCountry: string, titelCountry: string) => void;
};
