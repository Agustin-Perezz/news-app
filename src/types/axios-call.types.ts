import { AxiosResponse } from 'axios';

export type AxiosCall<T> = {
  call: Promise<AxiosResponse<T>>;
  controller?: AbortController;
};
