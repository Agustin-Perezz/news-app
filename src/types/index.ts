export * from './axios-call.types';
export * from './news-props';
export * from './fetch-data.types';
