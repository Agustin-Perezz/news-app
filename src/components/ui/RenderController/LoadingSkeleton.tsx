import {
  Box,
  Grid,
  Skeleton,
  Card,
  CardContent,
  CardMedia,
  Fade,
} from '@mui/material';
import { CustomGrid } from '../Cards';

export const LoadingSkeleton: React.FC = () => {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
      }}
    >
      <Grid sx={{ flexGrow: 1, margin: '25px 0px' }}>
        <Grid
          item
          xs={12}
          sx={{
            justifyContent: 'center',
            display: 'flex',
            position: 'relative',
          }}
        >
          <CustomGrid container spacing={3}>
            {Array(5)
              .fill(0)
              .map((__, index) => (
                // eslint-disable-next-line react/no-array-index-key
                <Grid key={index} item>
                  <Fade timeout={1500} in>
                    <Card sx={{ height: '448px', width: '278px' }}>
                      <CardMedia>
                        <Skeleton
                          variant="rectangular"
                          animation="wave"
                          width="100%"
                          height="170px"
                        />
                      </CardMedia>
                      <CardContent>
                        <Skeleton
                          variant="rounded"
                          animation="wave"
                          width="90%"
                          height="25px"
                          sx={{ marginTop: '10px' }}
                        />
                        <Skeleton
                          animation="wave"
                          width="100%"
                          height="18px"
                          sx={{ marginY: '5px', marginTop: '18px' }}
                        />
                        <Skeleton animation="wave" width="90%" height="18px" />
                        <Box
                          sx={{
                            display: 'flex',
                            justifyContent: 'flex-start',
                            gap: '5px',
                            marginTop: '15px',
                          }}
                        >
                          <Skeleton
                            animation="wave"
                            width="20%"
                            height="18px"
                          />
                          <Skeleton
                            animation="wave"
                            width="40%"
                            height="18px"
                          />
                        </Box>
                      </CardContent>
                    </Card>
                  </Fade>
                </Grid>
              ))}
          </CustomGrid>
        </Grid>
      </Grid>
    </Box>
  );
};
