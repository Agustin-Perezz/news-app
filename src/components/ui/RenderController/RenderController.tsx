import React from 'react';
import { NewsProps } from '../../../types/news-props';
import { ErrorBox } from '../Error';
import { LoadingSkeleton } from './LoadingSkeleton';

export type PropsRender = {
  children: React.ReactNode;
  isLoading: boolean;
  isError?: string;
  data?: NewsProps[];
};

export const RenderController: React.FC<PropsRender> = ({ children, data, isLoading, isError }) => {
  if (isLoading) return <LoadingSkeleton />;

  if (isError) return <ErrorBox errorMessage={isError} />;

  if (data?.length === 0) return <div>Ups, no results, try with others things.</div>;

  return <div>{children}</div>;
};
