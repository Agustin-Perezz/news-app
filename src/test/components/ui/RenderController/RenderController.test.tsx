import { render, screen } from '@testing-library/react';
import { PropsRender, RenderController } from '../../../../components/ui/RenderController';
import { NewsProps } from '../../../../types';

describe('Test in <RenderController />', () => {
  function setup(params: PropsRender) {
    render(<RenderController {...params}>{params.children}</RenderController>);
  }
  it('should match snapshot', () => {
    setup({ children: <span>test</span>, isLoading: false });
    expect(screen).toMatchSnapshot();
  });
  it('should render children', () => {
    setup({ children: <span>test</span>, isLoading: false });
    screen.getByText('test');
  });
  it('should not show content when isLoading is true', () => {
    setup({ children: <span>test</span>, isLoading: true });
    expect(screen.queryByText('test')).not.toBeInTheDocument();
  });
  it('should show error message', () => {
    setup({ children: <span>test</span>, isLoading: false, isError: 'Error' });
    screen.getByText('Error');
    expect(screen.queryByText('test')).not.toBeInTheDocument();
  });
  it('should information message when data length is equal to 0', () => {
    const data: NewsProps[] | undefined = [];
    setup({ children: <span>test</span>, isLoading: false, data });
    screen.getByText('Ups, no results, try with others things.');
  });
});
