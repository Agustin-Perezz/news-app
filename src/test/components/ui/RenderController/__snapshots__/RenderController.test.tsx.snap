// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RenderController /> should match snapshot 1`] = `
<div>
  <div>
    <span>
      test
    </span>
  </div>
</div>
`;
